criu/sockets.c

```diff
@@ -465,18 +466,35 @@ int do_restore_opt(int sk, int level, int name, void *val, int len)
 	return 0;
 }
 
-static int sk_setbufs(void *arg, int fd, pid_t pid)
+int sk_setbufs(int sk, uint32_t *bufs)
 {
-	u32 *buf = (u32 *)arg;
+	uint32_t sndbuf = bufs[0], rcvbuf = bufs[1];
 
-	if (restore_opt(fd, SOL_SOCKET, SO_SNDBUFFORCE, &buf[0]))
-		return -1;
-	if (restore_opt(fd, SOL_SOCKET, SO_RCVBUFFORCE, &buf[1]))
-		return -1;
+	if (setsockopt(sk, SOL_SOCKET, SO_SNDBUFFORCE, &sndbuf, sizeof(sndbuf)) ||
+	    setsockopt(sk, SOL_SOCKET, SO_RCVBUFFORCE, &rcvbuf, sizeof(rcvbuf))) {
+		if (opts.unprivileged) {
+			pr_info("Unable to set SO_SNDBUFFORCE/SO_RCVBUFFORCE, falling back to SO_SNDBUF/SO_RCVBUF\n");
+			if (setsockopt(sk, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) ||
+			    setsockopt(sk, SOL_SOCKET, SO_RCVBUF, &rcvbuf, sizeof(rcvbuf))) {
+				pr_perror("Unable to set socket SO_SNDBUF/SO_RCVBUF");
+				return -1;
+			}
+		} else {
+			pr_perror("Unable to set socket SO_SNDBUFFORCE/SO_RCVBUFFORCE");
+			return -1;
+		}
+	}
 
 	return 0;
 }
 
+static int sk_setbufs_ns(void *arg, int fd, pid_t pid)
+{
+	u32 *buf = (u32 *)arg;
```

Comment:
Can you please use uint32_t* for bufs here like in sk_setbufs. Also probably we don't need variable at all and can do something like `return sk_setbufs(fd, (uint32_t *)arg);` to shorten the code?